src/Microsoft.ML.StaticPipe/SgdStatic.cs

```diff
@@ -38,17 +39,55 @@ public static class SgdStaticExtensions
             Scalar<bool> label,
             Vector<float> features,
             Scalar<float> weights = null,
-            int maxIterations = Arguments.Defaults.MaxIterations,
-            double initLearningRate = Arguments.Defaults.InitLearningRate,
-            float l2Weight = Arguments.Defaults.L2Weight,
+            int maxIterations = Options.Defaults.MaxIterations,
+            double initLearningRate = Options.Defaults.InitLearningRate,
+            float l2Weight = Options.Defaults.L2Weight,
             ISupportClassificationLossFactory loss = null,
-            Action<Arguments> advancedSettings = null,
             Action<IPredictorWithFeatureWeights<float>> onFit = null)
         {
             var rec = new TrainerEstimatorReconciler.BinaryClassifier(
                 (env, labelName, featuresName, weightsName) =>
                 {
-                    var trainer = new StochasticGradientDescentClassificationTrainer(env, labelName, featuresName, weightsName, maxIterations, initLearningRate, l2Weight, loss, advancedSettings);
+                    var trainer = new StochasticGradientDescentClassificationTrainer(env, labelName, featuresName, weightsName, maxIterations, initLearningRate, l2Weight, loss);
+
+                    if (onFit != null)
+                        return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                    return trainer;
+
+                }, label, features, weights);
+
+            return rec.Output;
+        }
+
+        /// <summary>
+        ///  Predict a target using a linear binary classification model trained with the <see cref="Microsoft.ML.Trainers.StochasticGradientDescentClassificationTrainer"/> trainer.
+        /// </summary>
+        /// <param name="ctx">The binary classificaiton context trainer object.</param>
+        /// <param name="label">The name of the label column.</param>
+        /// <param name="features">The name of the feature column.</param>
+        /// <param name="weights">The name for the example weight column.</param>
+        /// <param name="options">Advanced arguments to the algorithm.</param>
```

Comment:
>arguments [](start = 43, length = 9)

options?

Reply:
#2154

---
In reply to: [248874445](https://github.com/dotnet/machinelearning/pull/2093#discussion_r248874445) [](ancestors = 248874445)

Reply:
But this is a new comment/new code. why not update it with this PR?

---
In reply to: [248874778](https://github.com/dotnet/machinelearning/pull/2093#discussion_r248874778) [](ancestors = 248874778,248874445)